src/main/java/core/basesyntax/service/GenerateReport.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.service;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class GenerateReport {
+
+    private String resultFileName;
```

Comment:
Consider parameterizing the file name to adhere to the Open/Closed Principle and enhance flexibility.